akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -179,7 +175,21 @@ object ShardRegion {
   /**
    * Reply to `GetCurrentRegions`
    */
-  @SerialVersionUID(1L) final case class CurrentRegions(regions: Set[Address]) {
+  @SerialVersionUID(2L) final case class CurrentRegions(regionActorPaths: Set[ActorPath]) {
```

Comment:
> no obvious way for the users to figure out where all regions are 

well, it's rather obvious (but not too easy for abuse), there is one region actor on each node (address) that is returned by CurrentRegions, so the user can send a message to that node to some user defined actor and then that actor can access the local regions via the extension as usual